lib/modules/reducer.js

```diff
@@ -22,7 +22,8 @@ export default (state=DEFAULT, action={}) => {
     }
 
     case actions.SET_PAGE: {
-      const referrerFromHistory = !isEmpty(state.currentPage)
+      const historyLen = state.history.length;
+      const referrerFromHistory = !isEmpty(state.currentPage) && historyLen > 1
```

Comment:
I think I would prefer that we address history later, citing time constraints, the priority of this issue, and that the authors are out until next year and I really don't have a good handle on how 2X/platform works at this level.